commit ed3ecbc8be313cb3613d6022548deb41750b7725
Author: Artem Polyakov <artpol84@gmail.com>
Date:   Tue Apr 20 15:16:24 2021 -0700

    pmix: Fix detection of Externally-built PMIx
    
    See https://github.com/open-mpi/ompi/issues/8823 for more details.
    
    Signed-off-by: Artem Polyakov <artpol84@gmail.com>

diff --git opal/mca/pmix/ext1x/pmix1x_component.c opal/mca/pmix/ext1x/pmix1x_component.c
index 747f95cc6139..57e91305fd85 100644
--- opal/mca/pmix/ext1x/pmix1x_component.c
+++ opal/mca/pmix/ext1x/pmix1x_component.c
@@ -95,8 +95,7 @@ static int external_component_query(mca_base_module_t **module, int *priority)
     char *t, *id;
 
     /* see if a PMIx server is present */
-    if (NULL != (t = getenv("PMIX_SERVER_URI")) ||
-	NULL != (id = getenv("PMIX_ID"))) {
+    if (NULL != (t = getenv("PMIX_NAMESPACE"))) {
 	/* if PMIx is present, then we are a client and need to use it */
 	*priority = 100;
     } else {
